session/session.go

```diff
@@ -1564,7 +1564,7 @@ func (s *session) ExecRestrictedStmt(ctx context.Context, stmtNode ast.StmtNode,
 	}()
 
 	if execOption.SnapshotTS != 0 {
-		se.sessionVars.SnapshotInfoschema, err = getSnapshotInfoSchema(s, execOption.SnapshotTS)
```

Comment:
Why `ExecRestrictedStmt` need to consider the temporary table?  it's usually used to execute some internal SQL statement, but no internal SQL will use the temporary table!
Can you ever construct a case that the `infoschema` obtained here visit the local temporary table?